lib/ClangImporter/ImportDecl.cpp

```diff
@@ -8555,6 +8559,14 @@ void ClangImporter::Implementation::collectMembersToAdd(
                                             members);
     }
 
+    // For NSObject, add the otherwise-hidden NSObject protocol. This is only
+    // used to import mirrored members.
+    // FIXME: Check for any class that conforms to the NSObject protocol?
```

Comment:
I believe NSProxy is the only other one in the Apple SDKs. We are absolutely terrible with it but if you wanted to avoid touching it maybe you could make the condition about root classes instead.

Reply:
The principled thing would be to check whether it's a root class that adopts the protocol in Objective-C.